Replace jsonip with icanhazip

t-richards opened the pull request:
## What

This PR replaces `jsonip.com` with http://icanhazip.com/, a free service which fulfills the same task, but does so without returning any extra data or junk solicitations.

See also: https://major.io/icanhazip-com-faq/

## Why 

We've noticed some significant issues with using the jsonip service. Requests error out or are blocked with frustrating frequency.

Fixes #13.

ben commented:
I know it's been a while, but I'd love to merge this. Any chance you can update the tests?

t-richards commented:
I tried updating the tests, but it looks like a lot of them are failing, even after reverting these proposed changes (and removing the IP address fetch altogether!). I'd be happy to fix up the IP address stuff if you can help me get the rest of the tests in working order.

Let's see what kind of trouble we can get into; See below for force-pushed commits:

t-richards committed:
Remove IP address fetch, fund account unconditionally

 - Hopefully this will help identify broken tests which are unrelated to IP address
fetching
 - Bump node versions in CI to test against only supported releases

ben commented:
Alright, I have pretty high confidence that this will work. Probably the tests are failing because Hubot's behavior has changed, and I don't have time to dig into it right now. I'll file a separate issue.
